src/jobs/java_isopod_deploy.yml

```diff
@@ -0,0 +1,79 @@
+description: >
+  Deploy java app image to environment using isopod.
+  Supports app-module from monorepo as well as single app from single-app-repo.
+
+parameters:
+  appname:
+    description: "Name of maven module for the app, or blank for single-app-repo"
+    type: string
+    default: ""
+  to:
+    description: "Kubernetes cluster to deploy to"
+    type: enum
+    enum: [ "dev", "prod"]
+    default: "dev"
+  predeploy_steps:
+    description: "Steps that will run before deployment"
+    type: steps
+    default: [ ]
+  postdeploy_steps:
+    description: "Steps that will run after deployment"
+    type: steps
+    default: [ ]
+  artifactory_username:
+    type: string
+    default: $DOCKER_JFROG_USERNAME
+  artifactory_password:
+    type: string
+    default: $DOCKER_JFROG_PASSWORD
+  isopod_version:
+    type: string
+    default: $ISOPOD_VERSION
+  slack_notify_success:
+    description: "Flag to send slack notification on success"
+    type: boolean
+    default: false
+  slack_ok_webhook:
+    type: string
+    default: $SLACK_WEBHOOK
+  slack_ok_branches:
+    description: "Comma separated list of branches for which notifications are sent"
+    type: string
+    default: "master"
+  slack_notify_failure:
+    description: "Flag to send slack notification on failure"
+    type: boolean
+    default: false
+  slack_fail_webhook:
+    type: string
+    default: $SLACK_WEBHOOK
+  slack_fail_branches:
+    description: "Comma separated list of branches for which notifications are sent"
+    type: string
+    default: "master"
+
+executor:
+  name: isopod
+  tag: << parameters.isopod_version >>
+  username: << parameters.artifactory_username >>
+  password: << parameters.artifactory_password >>
```

Comment:
@robert-nemet do you think this pseudo-polymorph executor can work?

if i validate the orb with `circleci orb pack --skip-update-check src > orb.yml ; circleci orb --skip-update-check validate orb.yml` i notice that

if `java_builder_docker` does not specify the `vm_image` parameter, validation fails with
```
Error: Error calling job: 'java_maven_build_test'
Error calling executor: 'java_builder_docker'
Unexpected argument(s): vm_image
```
however if i add the `vm_image` parameter there (dummy param, not used), validation does _NOT_ fail for `java_builder_vm`, which i would expect to have the same problem with the `java_version` parameter that is only present on `java_builder_docker`

TLDR: im a bit afraid that the resulting orb does not work for the `java_builder_vm`, and that the validation does not find the problem because it does not dynamically resolve the executor name and rather bakes something in the first time it happens to see a value 🤔